core/src/main/java/io/questdb/griffin/engine/functions/date/ToTimezoneIntervalFunctionFactory.java

```diff
@@ -0,0 +1,143 @@
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2024 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.date;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.sql.Function;
+import io.questdb.cairo.sql.Record;
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.BinaryFunction;
+import io.questdb.griffin.engine.functions.IntervalFunction;
+import io.questdb.std.IntList;
+import io.questdb.std.NumericException;
+import io.questdb.std.ObjList;
+import io.questdb.std.Numbers;
+import io.questdb.std.Misc;
+import io.questdb.std.datetime.microtime.TimestampFormatUtils;
+import io.questdb.std.datetime.microtime.Timestamps;
+import io.questdb.std.Interval;
+import org.jetbrains.annotations.NotNull;
+
+import static io.questdb.std.datetime.TimeZoneRuleFactory.RESOLUTION_MICROS;
+
+public class ToTimezoneIntervalFunctionFactory implements FunctionFactory {
+    @Override
+    public String getSignature() {
+        return "to_timezone(ΔS)";
+    }
+
+    @Override
+    public Function newInstance(
+            int position,
+            ObjList<Function> args,
+            IntList argPositions,
+            CairoConfiguration configuration,
+            SqlExecutionContext sqlExecutionContext
+    ) throws SqlException {
+        Function interval = args.getQuick(0);
+        Function timezone = args.getQuick(1);
+
+        if (timezone.isConstant()) {
+            return getIntervalFunction(argPositions, interval, timezone, 1);
+        } else {
+            return new ToTimezoneIntervalFunctionVar(interval, timezone);
+        }
+    }
+
+    @NotNull
+    static IntervalFunction getIntervalFunction(IntList argPositions, Function interval, Function timezone, int multiplier) throws SqlException {
+        final CharSequence tz = timezone.getStrA(null);
+        if (tz != null) {
+            final int hi = tz.length();
+            final long l = Timestamps.parseOffset(tz, 0, hi);
+            if (l == Long.MIN_VALUE) {
+                try {
+                    return new OffsetIntervalFunctionFromRules(

```

Comment:
two functions can be consolidated into one if you use `Timestamps.getTimezoneRules` . This function will generate rules for all possible timezone variations

Reply:
additionally, all function class names are typically suffixed with `Function`